eStore/Scenes/Theme/ThemeSettingsView.swift

```diff
@@ -6,4 +6,145 @@
 //  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
 //
 
-import Foundation
+import UIKit
+
+final class ThemeSettingsViewImpl: UIViewController {
+    var presenter: ThemeSettingsPresenter!
+    var theme: Theme!
+
+    private var themeSectionedMenu: SectionedMenuView!
+    private var colorPickerItems: [TintColorType] = []
+
+    override func loadView() {
+        view = UIView()
+
+        themeSectionedMenu = SectionedMenuView(frame: .zero, style: .grouped)
+        view.addSubview(themeSectionedMenu)
+        themeSectionedMenu.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Theme settings"
+        navigationItem.largeTitleDisplayMode = .never
+
+        themeSectionedMenu.tableFooterView = UIView() // Is needed to remove unnecessary separators
+        themeSectionedMenu.backgroundColor = .clear
+        themeSectionedMenu.theme = theme
+
+        apply(theme: theme)
+        presenter.handleLoadView()
+    }
+
+    private func apply(theme: Theme) {
+        view.backgroundColor = theme.backgroundColor
+        navigationController?.navigationBar.tintColor = theme.tintColor
+        navigationController?.navigationBar.barTintColor = theme.barColor
+        navigationController?.navigationBar.barStyle = theme.barStyle
+        navigationController?.navigationBar.titleTextAttributes = [.foregroundColor: theme.textColor]
+        navigationController?.navigationBar.largeTitleTextAttributes = [.foregroundColor: theme.textColor]
+        themeSectionedMenu.separatorColor = theme.borderColor
+    }
+}
+
+// MARK: - ThemeSettingsView implementation
+extension ThemeSettingsViewImpl: ThemeSettingsView {
+    func display(sections: [SectionedMenuSection]) {
+        themeSectionedMenu.sections = sections
+        themeSectionedMenu.reloadData()
+    }
+
+    func display(alert: Alert) {
+        let alertController = AlertFactory().make(alert: alert)
```

Comment:
Hey, that's a bad sign creating factories inside views. How do you inject another factory here otherwise? I would suggest injecting it, not creating.